.php_cs

```diff
@@ -38,46 +38,47 @@ $config->setRules(
     [
         '@PSR2' => true,
         'align_multiline_comment' => ['comment_type' => 'all_multiline'],
+        'array_indentation' => true,
         'array_syntax' => ['syntax' => 'short'],
         'blank_line_before_statement' => [
             'statements' => ['return', 'continue', 'break', 'declare', 'throw', 'try'],
         ],
-        'cast_spaces' => ['space' => 'none'],
-        'combine_consecutive_issets' => true,
-        'combine_consecutive_unsets' => true,
+        'binary_operator_spaces' => true,
+        'cast_spaces' => ['space' => 'single'],
+        'class_attributes_separation' => ['elements' => ['method']],
         'compact_nullable_typehint' => true,
         'concat_space' => ['spacing' => 'one'],
-        'dir_constant' => true,
-        'function_to_constant' => true,
+        'declare_equal_normalize' => true,
+        'explicit_indirect_variable' => true,
+        'explicit_string_variable' => true,
+        'fully_qualified_strict_types' => true,
         'function_typehint_space' => true,
         'heredoc_to_nowdoc' => true,
         'include' => true,
         'linebreak_after_opening_tag' => true,
         'list_syntax' => ['syntax' => 'short'],
         'lowercase_cast' => true,
         'magic_constant_casing' => true,
-        'mb_str_functions' => true,
+        'method_chaining_indentation' => true,
         'method_separation' => true,
-        'modernize_types_casting' => true,
+        'multiline_whitespace_before_semicolons' => true,
         'native_function_casing' => true,
-        'native_function_invocation' => true,
         'new_with_braces' => true,
         'no_blank_lines_after_class_opening' => true,
         'no_blank_lines_after_phpdoc' => true,
         'no_empty_comment' => true,
         'no_empty_phpdoc' => true,
         'no_empty_statement' => true,
-        'no_extra_consecutive_blank_lines' => true,
-        'no_homoglyph_names' => true,
+        'no_extra_blank_lines' => true,
         'no_leading_import_slash' => true,
         'no_leading_namespace_whitespace' => true,
-        'no_mixed_echo_print' => ['use' => 'print'],
+        'no_mixed_echo_print' => true,
         'no_multiline_whitespace_around_double_arrow' => true,
-        'no_multiline_whitespace_before_semicolons' => true,
-        'no_php4_constructor' => true,
```

Comment:
👎

Reply:
It's [deprecated as of PHP 7.0](http://php.net/manual/en/migration70.deprecated.php#migration70.deprecated.php4-constructors). This rule would be only useful if someone is porting a really old project.

Reply:
Yup - still don't like it... Still want people who've got their head stuck in the dark ages to be aware of it - still dont want E_DEPRECATED notices everywhere ;)

Reply:
I'd leave that off. Chances of anyone using this ruleset on that old a project are minuscule, but more importantly it may not produce the correct result:

> Risky when old style constructor being fixed is overridden or overrides parent one.

...in which case it's better that the maintainer of such project sees E_DEPRECATED.

Reply:
Meh, OK, it doesn't hurt. See 9e99adcc84316001f454931c8be14ef078a88af9.